sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts

```diff
@@ -94,28 +94,73 @@ export class MSFAANumberSharedService {
         APPLICATION_INVALID_DATA_TO_CREATE_MSFAA_ERROR,
       );
     }
-    return this.internalCreateMSFAANumber(
+    return this.internalActivateMSFAANumber(
       application.student.id,
       application.id,
       application.currentAssessment.offering.offeringIntensity,
       auditUserId,
     );
   }
 
+  /**
+   *
+   * @param studentId student for which the MSFAA record will be reactivated.
+   * @param referenceApplicationId reference application id.
+   * @param offeringIntensity offering intensity.
+   * @param auditUserId user causing the change.
+   * @param msfaaNumber msfaaNumber to be reactivated.
+   * @param dateSigned date the msfaaNumber got signed.
+   * @param serviceProviderReceivedDate serviceProviderReceivedDate date.
+   * @returns reactivated MSFAA record.
+   */
+  async reactivateMSFAANumber(
+    studentId: number,
+    referenceApplicationId: number,
+    offeringIntensity: OfferingIntensity,
+    auditUserId: number,
+    msfaaNumber: Partial<MSFAANumber>,
+    dateSigned: Date,
+    serviceProviderReceivedDate: Date,
+  ): Promise<MSFAANumber> {
+    return this.internalActivateMSFAANumber(
+      studentId,
+      referenceApplicationId,
+      offeringIntensity,
+      auditUserId,
+      {
+        msfaaNumberId: msfaaNumber.id,
+        msfaaNumber: msfaaNumber.msfaaNumber,
+        dateSigned,
+        serviceProviderReceivedDate,
+      },
+    );
+  }
+
   /**
    * Creates a new MSFAA record with a new number for the specified student.
    * @param studentId student to have a new MSFAA record created.
    * @param referenceApplicationId reference application id.
    * @param offeringIntensity offering intensity.
    * @param auditUserId user that should be considered the one that is causing the changes.
    * individually based on, for instance, the Part time/Full time.
+   * @param options.
+   * - `msfaaNumberId` id of the msfaaNumber to be reactivated.
+   * - `msfaaNumber` msfaaNumber to be reactivated.
+   * - `dateSigned` date the msfaaNumber got signed.
+   * - `serviceProviderReceivedDate` serviceProviderReceivedDate date.
    * @returns created MSFAA record.
    */
-  private async internalCreateMSFAANumber(
+  private async internalActivateMSFAANumber(
     studentId: number,
     referenceApplicationId: number,
     offeringIntensity: OfferingIntensity,
     auditUserId: number,
+    options?: {
```

Comment:
The options parameter is used to refer to a combination of parameters that are required and they would be maybe better expressed in a single object. Please see the below example as a suggestion.
```ts
private async internalActivateMSFAANumber(
  studentId: number,
  referenceApplicationId: number,
  offeringIntensity: OfferingIntensity,
  auditUserId: number,
  options?: {
    existingMSFAA: Pick<
      MSFAANumber,
      "id" | "msfaaNumber" | "dateSigned" | "serviceProviderReceivedDate"
    >;
  },
): Promise<MSFAANumber> {
```